server/model/telemetry.go

```diff
@@ -0,0 +1,51 @@
+package model
+
+import (
+	"fmt"
+
+	"server/config"
+
+	"github.com/segmentio/analytics-go/v3"
+	"gorm.io/gorm"
+)
+
+// event name - required for Amplitude
+const event string = "Deployment Completed"
+
+// This slice should have the same metrics as /server/model/enums.go DeploymentMetrics
+var metrics = []DeploymentMetric{
+	StartTime,
+	EndTime,
+	ApplicationId,
+	Region,
+	AccessType,
+	DeployStatus,
+	Errors,
+	Retries,
+}
+
+func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
```

Comment:
This function and the next should be in different package. The package `model` has code dealing with models that get persisted, yet this code is dealing with making calls to Segment.

Reply:
That was the idea behind creating a new package called telemetry in the previous PR because it was going to have its own enum, model and publisher. 
Do you suggest moving them all to the new package or just the telemetry.go functions?

Reply:
Moved telemetry.go to engine package as discussed